controllers/trial_job_controller.go

```diff
@@ -179,17 +183,24 @@ func (r *TrialJobReconciler) applyJobStatus(ctx context.Context, t *redskyv1beta
 	if matchingSelector, err := meta.MatchingSelector(job.Spec.Selector); err == nil {
 		podList := &corev1.PodList{}
 		if err := r.List(ctx, podList, client.InNamespace(job.Namespace), matchingSelector); err == nil {
+
 			// Look for pod failures (edge case where job controller doesn't update status properly, e.g. initContainer failure or unschedulable)
 			for i := range podList.Items {
 				s := &podList.Items[i].Status
 				if s.Phase == corev1.PodFailed {
-					trial.ApplyCondition(&t.Status, redskyv1beta1.TrialFailed, corev1.ConditionTrue, s.Reason, "", time)
+					trial.ApplyCondition(&t.Status, redskyv1beta1.TrialFailed, corev1.ConditionTrue, s.Reason, "trial pod failed", time)
 					dirty = true
 				}
+
 				// TODO We should consolidate this with `internal/ready/podFailed`
 				for _, c := range s.Conditions {
 					if c.Type == corev1.PodScheduled && c.Status == corev1.ConditionFalse && c.Reason == corev1.PodReasonUnschedulable {
-						trial.ApplyCondition(&t.Status, redskyv1beta1.TrialFailed, corev1.ConditionTrue, c.Reason, c.Message, time)
+						trial.ApplyCondition(&t.Status, redskyv1beta1.TrialFailed, corev1.ConditionTrue, c.Reason, fmt.Sprintf("trial pod: %s", c.Message), time)
+
+						// Patch the job and set parallelism to 0 to suspend the job and terminate any active pods
+						if err := r.Patch(ctx, job, client.RawPatch(types.StrategicMergePatchType, []byte(`{ "spec": { "parallelism": 0  } }`))); err != nil {
+							r.Log.Error(err, "unable suspend trial job", "job", job, "pod status", podList.Items[i].Status)
```

Comment:
Nit: the messages should be title case (unlike errors which should be lower case; unlike errors, the messages are presented directly). Also, I get nervous about putting full objects into the log; at this point I think it would be sufficient to have `WithValues("trial", fmt.Sprintf("%s/%s", t.Namespace, t.Name))` (matches what we do in the metric controller) and `WithValues("job", job.Name)`.